internal/version/release_version.go

```diff
@@ -40,15 +39,30 @@ type releaseVersionFetcher struct {
 	ctx context.Context
 }
 
-// LatestVersion encapsulates the logic to manage different cloud providers.
-func (s *releaseVersionFetcher) LatestVersion() (*ReleaseInformation, error) {
+// LatestWithCriteria retrieves the first release version that matches the criteria. We assume that ListReleases returns releases sorted by created_at value.
+func (s *releaseVersionFetcher) LatestWithCriteria(n int, matchCriteria func(tag string, tool string) bool, toolName string) (*github.RepositoryRelease, error) {
```

Comment:
[nit] let's please make `func(tag string, tool string) bool` a type, it would make reading this a bit easier